code/datums/perks/oddity.dm

```diff
@@ -499,3 +499,22 @@
 		split_phrase[index] = word
 
 	return sanitize(jointext(split_phrase," "))
+
+/datum/perk/njoy
+	name = "Njoy (Active)"
+	desc = "Your mind can focus on what is real, just like when you get rid of a painful earring."
+	icon_state = "cheerful"  //https://game-icons.net/1x1/lorc/cheerful.html
+
+	gain_text = "Your mind feels much more clear now."
+	lose_text = "You feel the shadows once more."
+
+/datum/perk/njoy/assign(mob/living/carbon/human/H)
+	if(..())
+		holder.sanity.positive_prob *= 0
+		holder.sanity.negative_prob *= 0
+		holder.sanity.insight_gain_multiplier *= 0.5
+
+/datum/perk/njoy/remove()
+	if(holder)
+		holder.sanity.insight_gain_multiplier *= 2
+	..()
```

Comment:
Looks like `remove()` should restore breakdown probabilities. `initial()` won't work because some perks and breakdowns modify the values. You could have this end the current breakdown on assign(), then store `positive_prob` and `negative_prob` in vars that get re-applied on `remove()`.

Reply:
Fixed it in a different way, thanks for pointing it out.